test/token/ERC721/ERC721.behavior.js

```diff
@@ -390,22 +404,24 @@ function shouldBehaveLikeERC721 (
 
       context('when the address that receives the approval is the owner', function () {
         it('reverts', async function () {
-          await shouldFail.reverting(
-            this.token.approve(owner, tokenId, { from: owner })
+          await shouldFail.reverting.withMessage(
+            this.token.approve(owner, tokenId, { from: owner }), 'ERC721: transfer to current owner'
           );
         });
       });
 
       context('when the sender does not own the given token ID', function () {
         it('reverts', async function () {
-          await shouldFail.reverting(this.token.approve(approved, tokenId, { from: other }));
+          await shouldFail.reverting.withMessage(this.token.approve(approved, tokenId, { from: other }),
+            'ERC721: approve caller is not owner nor approved');
         });
       });
 
       context('when the sender is approved for the given token ID', function () {
         it('reverts', async function () {
           await this.token.approve(approved, tokenId, { from: owner });
-          await shouldFail.reverting(this.token.approve(anotherApproved, tokenId, { from: approved }));
+          await shouldFail.reverting.withMessage(this.token.approve(anotherApproved, tokenId, { from: approved }),
+            'ERC721: approve caller is not owner nor approved');
```

Comment:
This message isn't correct, what is happening is that the approvee is already approved. We should split the `require`.